### PR TITLE
CLOUD-3506 Remove mod_cluster from OpenShift default configuration

### PR DESCRIPTION
**Pull request by rhusar**
Resolves
https://issues.redhat.com/browse/CLOUD-3506

### CODE REVIEW

**Comment by rhusar** [comment]
@jfdenise @jmesnil  Should we also remove this from the s2i? https://github.com/jboss-container-images/jboss-eap-modules/blob/master/jboss/container/eap/galleon/artifacts/opt/jboss/container/eap/galleon/eap-s2i-galleon-pack/src/main/resources/configs/standalone/standalone.xml/config.xml#L33

**Comment by jfdenise** [comment]
@rhusar , you made changes to a configuration file we keep for reference that has been generated from galleon. So yes, the place to remove it is: https://github.com/jboss-container-images/jboss-eap-modules/blob/master/jboss/container/eap/galleon/artifacts/opt/jboss/container/eap/galleon/eap-s2i-galleon-pack/src/main/resources/configs/standalone/standalone.xml/config.xml#L22-L33
I think @luck3y did similar changes in a branch.

**Comment by rhusar** [comment]
@jfdenise Given there is not a single readme or single instruction on how this works its not clear what is what.

**Comment by rhusar** [comment]
Hey @luck3y do you know what's holding this up? Thanks!

**Comment by luck3y** [comment]
@rhusar This change needs to be made at the configuration stage, which is generated not a static config. So we'll need to make that change there as well. I'll comment on the Jira.

**Comment by rhusar** [comment]
@luck3y Could you please point me to the sources of the "configuration stage" or any README or build script that would highlight their location?

**Comment by luck3y** [comment]
@rhusar See here for the default generation / addition: https://github.com/jboss-container-images/jboss-eap-modules/blob/master/jboss/container/eap/galleon/config/ee/artifacts/opt/jboss/container/eap/galleon/eap-s2i-galleon-pack/src/main/resources/configs/standalone/standalone.xml/config.xml#L70

**Comment by rhusar** [comment]
@luck3y That *IS* the file that I am changing in this PR.

**Comment by luck3y** [comment]
Oh, I just saw you pushed, thanks!

**Comment by rhusar** [comment]
@luck3y The change has always been there – but since the file moved it was just a clean rebase – so if you were looking for the new location, you would have only seen the old one, even though the rebase mechanism handled that cleanly.

**Comment by luck3y** [comment]
@rhusar I've marked the jira for CD20, so when @jmesnil and @jfdenise approve, it can be merged.

**Comment by rhusar** [comment]
@luck3y Ok, thanks!

File: jboss/container/config/cd/19.0/standalone-openshift.xml [hunk 1]
```diff
@@ -15,7 +15,6 @@
         <extension module="org.jboss.as.jsf"/>
         <extension module="org.jboss.as.logging"/>
         <extension module="org.jboss.as.mail"/>
-        <extension module="org.jboss.as.modcluster"/>
```
Comment (jfdenise) [inline jboss/container/config/cd/19.0/standalone-openshift.xml:18]: We didn't remove the modcluster in cd19. We should keep it in the reference file.

File: jboss/container/config/cd/19.0/standalone-openshift.xml [hunk 2]
```diff
@@ -460,13 +459,6 @@
         <subsystem xmlns="urn:wildfly:microprofile-health-smallrye:2.0" security-enabled="false" empty-liveness-checks-status="${env.MP_HEALTH_EMPTY_LIVENESS_CHECKS_STATUS:UP}" empty-readiness-checks-status="${env.MP_HEALTH_EMPTY_READINESS_CHECKS_STATUS:UP}"/>
         <subsystem xmlns="urn:wildfly:microprofile-jwt-smallrye:1.0"/>
         <subsystem xmlns="urn:wildfly:microprofile-metrics-smallrye:2.0" security-enabled="false" exposed-subsystems="*" prefix="${wildfly.metrics.prefix:jboss}"/>
-        <subsystem xmlns="urn:jboss:domain:modcluster:5.0">
```
Comment (jfdenise) [inline jboss/container/config/cd/19.0/standalone-openshift.xml:463]: We didn't remove the modcluster in cd19. We should keep it in the reference file.

Comment (mchoma) [reply jboss/container/config/cd/19.0/standalone-openshift.xml:463]: @rhusar @luck3y why this jfdenise objection was not addressed?

Comment (luck3y) [reply jboss/container/config/cd/19.0/standalone-openshift.xml:463]: @mchoma, it was, I think the git UI doesn't show it though: https://github.com/jboss-container-images/jboss-eap-modules/blob/master/jboss/container/config/cd/19.0/standalone-openshift.xml#L463

I had edited the PR to add those lines back to the config, then pushed it.

Comment (rhusar) [reply jboss/container/config/cd/19.0/standalone-openshift.xml:463]: LGTM, thanks again @luck3y

**Pull request merged**